File: app/src/main/java/com/bsas/androiddevs/manejoerrores/ui/activity/MainActivity.java
package com.bsas.androiddevs.manejoerrores.ui.activity;

import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.bsas.androiddevs.manejoerrores.R;
import com.bsas.androiddevs.manejoerrores.common.Movie;
import com.bsas.androiddevs.manejoerrores.presenter.MainPresenter;
import com.bsas.androiddevs.manejoerrores.presenter.PresenterProvider;
import com.bsas.androiddevs.manejoerrores.ui.activity.adapter.MoviesRecyclerViewAdapter;
import com.bsas.androiddevs.manejoerrores.ui.view.MainView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends BaseActivity<MainPresenter> implements MainView {

    private MoviesRecyclerViewAdapter moviesRecyclerViewAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        this.presenter.attachView(this);

        this.bindViews();
    }

    @Override
    protected void onResume() {
        super.onResume();
        this.presenter.getMovies();
    }

    @Override
    protected void onStop() {
        super.onStop();
        this.presenter.dettachView();
    }

    @Override
    protected MainPresenter createPresenter() {
        return PresenterProvider.get().getPresenter(this, MainPresenter.class);
    }

    @Override
    public void displayMovies(List<Movie> movies) {
        this.moviesRecyclerViewAdapter.refreshMovies(movies);
    }

    private void bindViews() {
        RecyclerView moviesRecyclerView = this.findViewById(R.id.movies_recycler_view);
        moviesRecyclerView.setHasFixedSize(true);
        RecyclerView.LayoutManager linearLayoutManager = new LinearLayoutManager(this);
        moviesRecyclerView.setLayoutManager(linearLayoutManager);

        this.moviesRecyclerViewAdapter = new MoviesRecyclerViewAdapter(new ArrayList<Movie>());
        moviesRecyclerView.setAdapter(this.moviesRecyclerViewAdapter);
    }
}

File: app/src/main/java/com/bsas/androiddevs/manejoerrores/ui/activity/MainActivity.kt
package com.bsas.androiddevs.manejoerrores.ui.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.LinearLayoutManager
import com.bsas.androiddevs.manejoerrores.R
import com.bsas.androiddevs.manejoerrores.common.Movie
import com.bsas.androiddevs.manejoerrores.databinding.ActivityMainBinding
import com.bsas.androiddevs.manejoerrores.ui.activity.adapter.MoviesRecyclerViewAdapter
import com.bsas.androiddevs.manejoerrores.viewmodel.MainViewModel

class MainActivity : AppCompatActivity() {

    private lateinit var viewModel : MainViewModel
    private lateinit var binding : ActivityMainBinding

    private lateinit var moviesRecyclerViewAdapter : MoviesRecyclerViewAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        this.binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        this.viewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)

        this.bindObservers()
        this.bindRecyclerAdapter()
    }

    private fun bindObservers() {
        this.viewModel.movies.observe(this, Observer { this.refreshMovieList(it) })
    }

    private fun refreshMovieList(movies: List<Movie>) {
        this.moviesRecyclerViewAdapter.refreshMovies(movies)
        this.binding.invalidateAll() //Neccesary?
    }

    private fun bindRecyclerAdapter() {
        val moviesRecyclerView = this.binding.moviesRecyclerView
        moviesRecyclerView.setHasFixedSize(true)

        val linearLayoutManager = LinearLayoutManager(this)
        moviesRecyclerView.layoutManager = linearLayoutManager

        this.moviesRecyclerViewAdapter = MoviesRecyclerViewAdapter(listOf())
        moviesRecyclerView.adapter = this.moviesRecyclerViewAdapter
    }


}